web/src/main/java/ua/com/vertex/dao/CourseDaoImpl.java

```diff
@@ -49,4 +82,168 @@ public CourseDaoImpl(DataSource dataSource) {
+        mapSqlParameterSource.addValue(COLUMN_EMAIL, dto.getEmail());
+        mapSqlParameterSource.addValue(COLUMN_FIRST_NAME, dto.getFirstName());
+        mapSqlParameterSource.addValue(COLUMN_LAST_NAME, dto.getLastName());
+        mapSqlParameterSource.addValue(COLUMN_PHONE, dto.getPhone());
+
+        jdbcTemplate.update(query, mapSqlParameterSource);
+
+        LOGGER.debug("User was assigned to the course");
+    }
+
+    @Override
+    public List<User> searchForUsersToAssign(CourseUserDto dto) {
+        LOGGER.debug(String.format(logInfo.getId() + "Searching for users to assign to the course by search param=%s",
+                dto.getSearchParam()));
+
+        List<User> users;
+        String query = "";
+        switch (dto.getSearchType()) {
+            case "first_name":
+                query = "SELECT u.user_id, u.email, u.first_name, u.last_name, u.phone FROM Users u " +
+                        "WHERE u.first_name LIKE '%" + dto.getSearchParam() + "%' AND " +
+                        "(SELECT count(*) FROM Course_users c WHERE c.email=u.email AND c.course_id=:courseId) = 0";
+                break;
+            case "last_name":
+                query = "SELECT u.user_id, u.email, u.first_name, u.last_name, u.phone FROM Users u " +
+                        "WHERE u.last_name LIKE '%" + dto.getSearchParam() + "%' AND " +
+                        "(SELECT count(*) FROM Course_users c WHERE c.email=u.email AND c.course_id=:courseId) = 0";
+                break;
+            case "email":
+                query = "SELECT u.user_id, u.email, u.first_name, u.last_name, u.phone FROM Users u " +
+                        "WHERE u.email LIKE '%" + dto.getSearchParam() + "%' AND " +
+                        "(SELECT count(*) FROM Course_users c WHERE c.email=u.email AND c.course_id=:courseId) = 0";
+        }
+        MapSqlParameterSource mapSqlParameterSource = new MapSqlParameterSource();
+        mapSqlParameterSource.addValue(SEARCH_PARAM, dto.getSearchParam());
+        mapSqlParameterSource.addValue(COLUMN_COURSE_ID2, dto.getCourseId());
+
+        users = jdbcTemplate.query(query, mapSqlParameterSource, this::mapUser);
+
+        LOGGER.debug(logInfo.getId() + "Retrieved users id=(" + userIdsToString(users) + ")");
+
+        return users;
+    }
+
+    private User mapUser(ResultSet resultSet, int i) throws SQLException {
+        return new User.Builder()
+                .setUserId(resultSet.getInt("user_id"))
+                .setEmail(resultSet.getString("email"))
+                .setFirstName(resultSet.getString("first_name"))
+                .setLastName(resultSet.getString("last_name"))
+                .setPhone(resultSet.getString("phone"))
+                .getInstance();
+    }
+
+    private String userIdsToString(List<User> users) {
+        return users.stream().map(User::getUserId).map(String::valueOf).collect(Collectors.joining(", "));
```

Comment:
No, they are completely different. The code in your previous comment logs just a single value. In my case the code logs a list of values, that's why I use lambda to process the values and transform them to the required form.